python/ec2_require_security_group_by_tag.py

```diff
@@ -0,0 +1,71 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+# Ensure all EC2 Instances that have a certain tag format also have a specific security group
+# Description: Checks that all EC2 instances that have a certain tag format also have a specific security group
+#
+# Trigger Type: Change Triggered
+# Scope of Changes: EC2:Instance
+# Required Parameters: name
+# Example Value: ^prod(us|eu|br)[lw]box[0-9]{3}$ (which will match 'produslbox001')
+# Required Parameters: secGroup
+# Example Value: MySecGroup
+# 
+
+import boto3
+import json
+import re
+
+def is_applicable(config_item, event):
+    status = config_item['configurationItemStatus']
+    event_left_scope = event['eventLeftScope']
+    test = ((status in ['OK', 'ResourceDiscovered']) and
+            event_left_scope == False)
+    return test
+
+def evaluate_compliance(config_item, rule_parameters):
+    # Initialize evaluation to 'not applicable', i.e. rule doesn't apply
+    evaluation = 'NOT_APPLICABLE'
+    if (config_item['resourceType'] == 'AWS::EC2::Instance'):
+        configuration = config_item['configuration']
+        tags = configuration['tags']
+        foundTag = False
+        # If the config item is for an EC2 instance, then iterate through the tags for that instance
+        for tag in tags:
+            # Check if this is the name tag, and that it matches the provided regex value
```

Comment:
[nit] - "Check if this is the 'Name' tag..." or "Check if this tag has key 'Name'..." may be more readable